x-pack/plugins/profiling/server/routes/downsampling.ts

```diff
@@ -61,27 +61,33 @@ export function getSampledTraceEventsIndex(
   };
 }
 
-export async function findDownsampledIndex(
-  logger: Logger,
-  client: ElasticsearchClient,
-  index: string,
-  filter: ProjectTimeQuery,
-  sampleSize: number
-): Promise<DownsampledEventsIndex> {
+export async function findDownsampledIndex({
+  logger,
+  client,
+  index,
+  filter,
+  sampleSize,
+}: {
+  logger: Logger;
+  client: ProfilingESClient;
+  index: string;
+  filter: ProjectTimeQuery;
+  sampleSize: number;
```

Comment:
A general question for my understanding. This pattern introduces code redundancy and is not as readable as before. So it must have some advantage over the pattern on the left side that I don't see. Can you give me some background ?

Reply:
Why do you feel it introduces code redundancy compared to the previous approach? because of the syntax? The advantage is that you can add or remove parameters without much risk. This is a contrived example, but if you swap two arguments around of the same type, perhaps because you want to make the last one optional, there's no way TypeScript can see that you should update calls to the function to also swap those around. In terms of readability, I think it also is the better option, because you can more easily see from how the function is called what every argument means.

Reply:
Suppose I have the following function definition:

```ts
function calculateANumber ( min:number, max:number ) {}

const myMin = 0;
const myMax = 10;

calculateANumber(myMin, myMax);
```

suppose I change the signature:

```ts
function calculateANumber ( max:number, min:number =0 ) {}

const myMin = 0;
const myMax = 10;

calculateANumber(myMin, myMax)
```

TypeScript won't warn me, and there isn't a quick way to tell from the calling code that this is now wrong. However, if I use the following syntax: 

```ts
function calculateANumber ( { min, max }:{ min:number, max:number } ) {}

const myMin = 0;
const myMax = 10;

calculateANumber({ min: myMin, max: myMax});
```

I can easily add a default value, and my code will still be correct:

```ts
function calculateANumber ( { min = 0, max }:{ min:number, max:number } ) {}

const myMin = 0;
const myMax = 10;

calculateANumber({ min: myMin, max: myMax});
```

Does that make sense?

Reply:
re:
> Maybe this syntax is idiomatic TS ?

it's idiomatic for the code I've seen in Kibana. It's also commonly enforced (but not in Kibana) by a lint rule: https://eslint.org/docs/latest/rules/max-params.

Reply:
> Does that make sense?

Thank you, yes. Due to my lack of JS/TS practice I didn't recognize 'argument list' vs. 'object as argument'. Now seeing the function call using an object and the lights went on 😄